### ISSUE TITLE
'Stat' API method and wiki documention

### ISSUE THREAD

**Issue by wagnered** [ampache/ampache#2255]
In the wiki conflict seems to exist between the initial description of 'stat' and the description of the 'filter' parameter.  Filter can be null vs filter not optional.

Shouldn't the filter type be added to the result, especially if the 'random' type was selected?

**Comment by lachlan-00** [comment]
The old stats method didn't require input to fall back to random but I would like this method to have a filter so you know what request you're sending in an obvious way.

That's why I've documented as mandatory in the wiki. Null filter is really only there as backward compatibility with 3.8 API calls

**Comment by wagnered** [comment]
> The old stats method didn't require input to fall back to random   I infer from that "random" was the default filter type prior to 4.0. The following is from what I believe is the viewpoint of a casual(newbie) user.  By the way, this is the viewpoint everyone should take who contributes to a user-oriented document. What I read is:  confusing:  The description says it's optional, the table says it's not. What it should read is: "The default filter type has been deprecated and will be removed sometime in the future."  Until that time or version, the filter parameter should remain specified as optional.  This should be spelled out in a timeline, possibly on a project page.  This gives the user an advanced notice.  Changes should not be made on a whim and without proper documentation.  I just recently discovered the Ampache developer's team group. It is in this group that every potential change should be discussed and adapted by consensus or by majority.  The "to-do" or specs can then be laid out in the project. As for Ampache specs, they should not be defined in a wiki(how-to) document. Doing so reduces the legitimacy of the entire Ampache project. If any of the 8 collaborators has good organizational skills, I think that now would be a good time to reorganize the docs.  For example, separate out any Ampache technical documentation from the wiki docs and store them in a separate repository. Possibly in ampache.org, (ampache.github.io) and reference them  from the wiki and other docs. Even though Ampache is open source, it should maintain an appearance of 0professionalism.

**Comment by wagnered** [comment]
Sorry, i copied text from my email editor and pasted it into the issues web page.  I couldn't delete the comment. Hopefully, I'll have a new tablet soon.

> The old stats method didn't require input to fall back to random 

 I infer from that the filter parameter was optional and the default filter type prior to 4.0 was "random".

The following is from, what I believe, the viewpoint of a casual(newbie) user.  By the way, this is the viewpoint everyone should take when contributing to a user-oriented document.

What I read is:  Confusing:  The description says it's optional, the table says it's not.

What it should read is something like this: "The default filter type has been deprecated and will <probably> be removed sometime <or in a specific version> in the future."  Until that time or version, the filter parameter should remain specified as optional.  This should be spelled out in a timeline, possibly on a project page.  This gives the user an advanced notice.  It is also not ambiguous.

Changes should not be made on a whim and without proper documentation.  I just recently discovered the Ampache developer's team group. It is in this group that every potential change should be discussed and adapted by consensus or by majority.  The "to-do" or specs can then be laid out in the project.

As for Ampache specs, they should not be defined in a wiki(how-to) document. Doing so reduces the legitimacy of the entire Ampache project.

If any of the 8 collaborators has good organizational skills, I think that now would be a good time to reorganize the docs.  For example, separate out any Ampache technical documentation from the wiki docs and store them in a separate repository. Possibly in ampache.org, (ampache.github.io) and reference them  from the wiki and other docs.

Even though Ampache is open source, it should maintain an appearance of professionalism.null

**Comment by lachlan-00** [comment]
I've actually been planning this for a while, I'm just not able to pay for my plans yet. I've used the wiki as the primary source to get things on paper. 

I'm getting together money to provide a semi-official official demo site for API Devs.

I'm also going to put out bounties for website work to attract some new people.

Then integrate all the technical parts of the wiki into the website / demo site.

2019 has been a code-first process but I've been putting a lot of thought into how to make 2020 an even bigger year. :)

**Comment by kuzi-moto** [comment]
> I'm getting together money to provide a semi-official official demo site for API Devs.

What would you like it to do ideally? I can use my host for the user demo for that as well. Unless you'd like more control over it of course.

**Comment by wagnered** [comment]
I managed to procure a Galaxy tab A 10.1 from Costco on sale 😃 .  I'm now setting up a development environment.


I noticed that the donate buttons on ampache.org and every installation points to @afterster's PayPal account and credit card contributions.  I would have no problem if you redirected them to your own accounts. Or you going to set up a crowd funding source 😄 ? Or both?

**Comment by lachlan-00** [comment]
Personally I'm wary of donations. I'm happy to save a bit and put thing up myself. I've also applied for a grant to fund a couple of years of hardware/hosting so it's actually not a huge cost personally.

When there is something to actually donate for I might want to do that but I have to do the work first and I'm happy to pay for the gap on things to keep it free and open

**Comment by lachlan-00** [comment]
I've started moving the wiki pages for the api into flatdoc http://ricostacruz.com/flatdoc/. This will then go into ampache.org

seems easy enough and doesn't look terrible either. I like it's layout and the design is easier to navigate than the wiki

**Comment by wagnered** [comment]
Is there a way to sync changes with changes to the Ampache wiki? I was thinking of cloning the wiki and do some editing? Or will the github wiki be removed?
-- 
Sent from my Android device with Email Mail. Please excuse my brevity.

**Comment by lachlan-00** [comment]
Flatdoc can point to a url so I think it's able to stay. I did get some crossdomain lerrors in internal testing so we'll see if it needs to go later. Right now I'm just playing with it but I like it enough that I'll start theming it as well.

Tldr, edit away. It's the current point of truth

**Comment by lachlan-00** [comment]
Let me know when you've made the wiki changes you wanted to do. i've got a themed header right now and will start working on a master api markdown file once you're done.
![image](https://user-images.githubusercontent.com/1305249/71859208-cc609500-3139-11ea-8050-d136c82fc444.png)

What i really like about this theme is the 3 columns provide space for responses inline with the text
![image](https://user-images.githubusercontent.com/1305249/71859246-f619bc00-3139-11ea-921f-a1ee419d3460.png)

**Comment by wagnered** [comment]
@lachlan-00 Sorry  to say that my circumstances has changed.  I'm not sure how many of the team knows my age (72) but I'm not in the best of health. Medical tests, procedures, and other obligations prevent me from following through.  So carry on. It looks good.

I'm probably eligible for the Guiness book of records for the oldest active 'coder'.  Although I prefer to think of myself as a programmer.


null

**Comment by lachlan-00** [comment]
You know you might be right there!

Don't stress about this or us then. It's been really good to have you back though so stick around as much as you can.

I have updated the stat docs and I think I'll keep things as a strict "current version" docstring with info about changes in the comment.
That way the documentation always show the way things are expected without confusing previous iterations

**Comment by AshotN** [comment]
`Stats` will not work without the `filter` field. If we are trying to preserve backwards compatibility, it should be made optional. With perhaps a notice that it will become mandatory soon.

**Comment by lachlan-00** [comment]
It's still optional in the code but isn't documented as optional.

When I get a chance I'm going to write up a list of functions for the API vs subsonic and then API vs web function.

That way we can map this thing out fully with a defined process for API Devs on how to do things.

**Comment by lachlan-00** [comment]
it's optional now the back-compat has been removed in develop
